clippy_lints/src/unused_async.rs

```diff
@@ -81,6 +81,20 @@ impl<'a, 'tcx> Visitor<'tcx> for AsyncFnVisitor<'a, 'tcx> {
     }
 }
 
+/// Checks if the `def_id` belongs to a function and that function is part of a trait impl,
+/// in which case we shouldn't lint because `async` is part of the trait definition and therefore
+/// can't be removed.
+fn is_in_trait_impl(cx: &LateContext<'_>, def_id: LocalDefId) -> bool {
```

Comment:
issue: this should be in utils somewhere. there's also a chance we already have it

(@Centri3 got opinions where?)

Reply:
I don't think we already have it, we have some for `MethodCall`s and such but not `DefId`s themselves

I think just the top-level `clippy_utils` is fine. The other `is_trait_method` function is there too. We could probably rename this to `is_def_id_trait_method` or something for consistency's sake.